AdvancementAPI.java

```diff
@@ -13,27 +15,37 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Paths;
+import java.util.Collection;
 import java.util.List;
 import com.google.common.collect.Lists;
 
 /**
- * Created by charliej on 14/05/2017.
- * Last modification DiscowZombie on 5/06/2017.
+ * Rewritten by PROgrammer_JARvis
+ * Originally Created by charliej on 14/05/2017.
+ * Lately modified by DiscowZombie on 5/06/2017.
  */
 
-public class AdvancementAPI {
+//Renamed into MessengerAdvancement as AdvancementAPI seems to be a better name for some Singleton @PROgrm_JARvis
+public class MessengerAdvancement {
 
     private NamespacedKey id;
-    private String title, parent, trigger, icon, description, background;
-    private boolean announce, toast;
-    private FrameType frame;
-    private List<ItemStack> items;
-
-    public AdvancementAPI(NamespacedKey id) {
+    private String
+            title = "Untitled",
+            parent,
+            trigger = "minecraft:impossible",
+            icon = "minecraft:golden_apple",
+            description = "no description",
+            background = "minecraft:textures/gui/advancements/backgrounds/stone.png";
+    private boolean announce = false, toast = true;
+    private FrameType frame = FrameType.TASK;
+    private List<ItemStack> items = Lists.newArrayList();
+
+    public MessengerAdvancement(NamespacedKey id) {
         this.id = id;
-        this.items = Lists.newArrayList();
-        this.announce = true;
-        this.toast = true;
+    }
+
+    public MessengerAdvancement(String id) {
+        this.id = new NamespacedKey(ContentMakersPlugin.getInstance(), id);
```

Comment:
Won't work this way. You've to pass the the JavaPlugin. `ContentMakersPlugin.getInstance()` only works with your plugin.